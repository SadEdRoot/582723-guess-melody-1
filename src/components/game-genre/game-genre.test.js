import React from 'react';
import GameGenre from './game-genre.jsx';
import renderer from 'react-test-renderer';

const mocks = {
  "type": `genre`,
  "genre": `reggae`,
  "answers": [
    {
      "src": `https://es31-server.appspot.com/guess-melody/static/music/Addis_Ababa.mp3`,
      "genre": `reggae`
    },
    {
      "src": `https://es31-server.appspot.com/guess-melody/static/music/Azure.mp3`,
      "genre": `electronic`
    },
    {
      "src": `https://es31-server.appspot.com/guess-melody/static/music/Whaling_City.mp3`,
      "genre": `country`
    },
    {
      "src": `https://es31-server.appspot.com/guess-melody/static/music/Skanada.mp3`,
      "genre": `reggae`
    }
  ]
};

it(`GameGenre correctly renders`, () => {
  const tree = renderer.create(<GameGenre
    gameTime={0}
    errorCount={0}
    question={mocks}
    onSubmit={()=>{}}
  />)
  .toJSON();

  expect(tree).toMatchSnapshot();
});
