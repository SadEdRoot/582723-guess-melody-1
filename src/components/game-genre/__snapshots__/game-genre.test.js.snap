// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameGenre correctly renders 1`] = `
<section
  className="game game--genre"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="390"
        cy="390"
        filter="url(#blur)"
        r="370"
        style={
          Object {
            "MsTransformOrigin": "center",
            "WebkitTransformOrigin": "center",
            "transformOrigin": "center",
          }
        }
      />
    </svg>
    <div
      className="timer__value"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <span
        className="timer__mins"
      >
        0
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    />
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Выберите 
      reggae
       треки
    </h2>
    <form
      className="game__tracks"
      onSubmit={[Function]}
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            id="artist-0"
            name="answer"
            type="checkbox"
            value="artist-0"
          />
          <label
            className="game__check"
            htmlFor="artist-0"
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            id="artist-1"
            name="answer"
            type="checkbox"
            value="artist-1"
          />
          <label
            className="game__check"
            htmlFor="artist-1"
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            id="artist-2"
            name="answer"
            type="checkbox"
            value="artist-2"
          />
          <label
            className="game__check"
            htmlFor="artist-2"
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            id="artist-3"
            name="answer"
            type="checkbox"
            value="artist-3"
          />
          <label
            className="game__check"
            htmlFor="artist-3"
          >
            Отметить
          </label>
        </div>
      </div>
      <button
        className="game__submit button"
        type="submit"
      >
        Ответить
      </button>
    </form>
  </section>
</section>
`;
